[pt-br] Add docs/tasks/administer-cluster/kubelet-credential-provider.md (#40547)
kubernetes/website: content/pt-br/docs/tasks/administer-cluster/kubelet-credential-provider.md

---

> title: Configurar um provedor de credenciais de imagem kubelet.

Reviewer: ```suggestion
title: Configurar um provedor de credenciais de imagem para o kubelet
```

Very minor suggestion for readability.

reviewers:
- liggitt

> - cheftako

Reviewer: The reviewers section can be removed entirely.

Reviewer: Hi @ricmmartins, the reviewers section is still here. It must be removed.

> description: Configure o plugin de provedor de credenciais de imagem do kubelet.

Reviewer: ```suggestion
description: Configure o plugin de provedor de credenciais de imagem do kubelet
```

Original text had no period.

content_type: task
min-kubernetes-server-version: v1.26
weight: 120
---

{{< feature-state for_k8s_version="v1.26" state="stable" >}}

<!-- overview -->

> A partir do Kubernetes v1.20, o kubelet pode recuperar dinamicamente as credenciais para um registro de imagem de contêiner usando plugins executáveis. O kubelet e o plugin de execução se comunicam por meio de stdio (stdin, stdout e stderr) usando APIs versionadas do Kubernetes. Esses plugins permitem que o kubelet solicite credenciais para um registro de contêiner dinamicamente, em vez de armazenar credenciais estáticas no disco. Por exemplo, o plugin pode se comunicar com um servidor de metadados local para recuperar credenciais de curta duração para uma imagem que está sendo puxada pelo kubelet.

Reviewer: ```suggestion
A partir do Kubernetes v1.20, o kubelet pode obter dinamicamente as credenciais para um registro de imagem de contêiner usando plugins executáveis. O kubelet e o plugin executável se comunicam por meio de stdio (stdin, stdout e stderr) usando APIs versionadas do Kubernetes. Esses plugins permitem que o kubelet solicite credenciais para um registro de contêiner dinamicamente, em vez de armazenar credenciais estáticas no disco. Por exemplo, o plugin pode se comunicar com um servidor de metadados local para recuperar credenciais de curta duração para uma imagem que está sendo baixada pelo kubelet.
```

Some minor suggestions in this paragraph. Main one is keeping consistency between your translations for "exec plugin".

> Você pode estar interessado em usar essa capacidade se alguma das condições abaixo for verdadeira:

Reviewer: Is there a better word you could use here instead of "capacidade"?
Perhaps "funcionalidade"? I feel like "capability" doesn't translate very well to Portuguese.

Author: agree that "funcionalidade" sounds better

* Chamadas de API para um serviço de provedor de nuvem são necessárias para recuperar informações de autenticação para um registro.
* As credenciais têm tempos de expiração curtos e é necessário solicitar novas credenciais com frequência.
* Armazenar credenciais de registro no disco ou em `imagePullSecrets` não é aceitável.

Este guia demonstra como configurar o mecanismo de plugin do provedor de credenciais de imagem do kubelet.

## {{% heading "prerequisites" %}}

* Você precisa de um cluster Kubernetes com nós que suportem plugins de provedor de credenciais do kubelet. Esse suporte está disponível no Kubernetes {{< skew currentVersion >}}; As versões v1.24 e v1.25 do Kubernetes incluíram isso como um recurso beta, ativado por padrão.

> * Uma implementação funcional de um plugin exec de provedor de credenciais. Você pode construir seu próprio plugin ou usar um fornecido por provedores de nuvem.

Reviewer: ```suggestion
* Uma implementação funcional de um plugin executável de provedor de credenciais. Você pode criar seu próprio plugin ou usar um fornecido por provedores de nuvem.
```
You used "plugin executável" before. Let's keep it consistent throughout the page.

{{< version-check >}}

<!-- steps -->

## Instalando Plugins nos Nós

Um plugin de provedor de credenciais é um binário executável que será executado pelo kubelet. Certifique-se de que o binário do plugin exista em cada nó do seu cluster e armazenado em um diretório conhecido. O diretório será necessário posteriormente ao configurar as flags do kubelet.

## Configurando o Kubelet

Para usar esse recurso, o kubelet espera que duas flags sejam definidas:

* `--image-credential-provider-config` - o caminho para o arquivo de configuração do plugin de provedor de credenciais.
* `--image-credential-provider-bin-dir` - o caminho para o diretório onde estão localizados os binários do plugin de provedor de credenciais.

### Configurar um provedor de credenciais do kubelet

> O arquivo de configuração passado para `--image-credential-provider-config` é lido pelo kubelet para determinar quais plugins exec devem ser invocados para quais imagens de contêiner. Aqui está um exemplo de arquivo de configuração que você pode acabar usando se estiver usando o plugin baseado no [ECR](https://aws.amazon.com/ecr/):

Reviewer: ```suggestion
O arquivo de configuração passado para `--image-credential-provider-config` é lido pelo kubelet para determinar quais plugins executáveis devem ser invocados para quais imagens de contêiner. Aqui está um exemplo de arquivo de configuração que você pode acabar usando se estiver usando o plugin baseado no [ECR](https://aws.amazon.com/ecr/):
```

Same bit on consistency here.

```yaml
apiVersion: kubelet.config.k8s.io/v1
kind: CredentialProviderConfig
# providers is a list of credential provider helper plugins that will be enabled by the kubelet.
# Multiple providers may match against a single image, in which case credentials
# from all providers will be returned to the kubelet. If multiple providers are called
# for a single image, the results are combined. If providers return overlapping

> # auth keys, the value from the provider earlier in this list is used.

Reviewer 2: What do you think to translate the comments?

providers:
   # name é o nome necessário do provedor de credenciais. Deve corresponder ao nome do
   # executável do provedor visto pelo kubelet. O executável deve estar no 

>    # diretório bin do kubelet (definido pelo sinalizador --image-credential-provider-bin-dir).

Reviewer: ```suggestion
   # diretório bin do kubelet (definido pela flag --image-credential-provider-bin-dir).
```
Very minor, mainly for consistency because you used flag before.

   - name: ecr
     # matchImages é uma lista obrigatória de strings usadas para corresponder às imagens para
     # determinar se este provedor deve ser invocado. Se uma das strings corresponder à
     # imagem solicitada do kubelet, o plug-in será invocado e terá uma chance
     # para fornecer credenciais. Espera-se que as imagens contenham o domínio de registro
     # e caminho da URL.
     #
     # Cada entrada em matchImages é um padrão que pode opcionalmente conter uma porta e um caminho.
     # Globs podem ser usados no domínio, mas não na porta ou no caminho. Globs são suportados
     # como subdomínios como '*.k8s.io' ou 'k8s.*.io' e domínios de nível superior como 'k8s.*'.
     # A correspondência de subdomínios parciais como 'app*.k8s.io' também é suportada. Cada glob só pode corresponder
     # a um único segmento de subdomínio, então `*.io` **não** corresponde a `*.k8s.io`.
     #
     # Existe uma correspondência entre uma imagem e uma matchImage quando todas as opções abaixo são verdadeiras:
     # - Ambos contêm o mesmo número de partes de domínio e cada parte faz correspondência.

>      # - O caminho do URL de um matchImages deve ser um prefixo do caminho do URL da imagem de destino.

Reviewer: ```suggestion
     # - O caminho da URL de um matchImages deve ser um prefixo do caminho do URL da imagem de destino.
```

You wrote "da URL" in a previous paragraph.

     # - Se matchImages contiver uma porta, a porta também deverá corresponder à imagem.
     #
     # Valores de exemplo de matchImages:
     # - 123456789.dkr.ecr.us-east-1.amazonaws.com
     # - *.azurecr.io
     # - gcr.io
     # - *.*.registry.io
     # - Registry.io:8080/path
     matchImages:
       - "*.dkr.ecr.*.amazonaws.com"
       - "*.dkr.ecr.*.amazonaws.cn"
       - "*.dkr.ecr-fips.*.amazonaws.com"
       - "*.dkr.ecr.us-iso-east-1.c2s.ic.gov"
       - "*.dkr.ecr.us-isob-east-1.sc2s.sgov.gov"
     # defaultCacheDuration é a duração padrão em que o plug-in armazenará as credenciais na memória
     # se a duração do cache não for fornecida na resposta do plug-in. Este campo é obrigatório.
     defaultCacheDuration: "12h"
     # Versão de entrada necessária do exec CredentialProviderRequest. O CredentialProviderResponse retornado
     # DEVE usar a mesma versão de codificação da entrada. Os valores atualmente suportados são:
     # - credentialprovider.kubelet.k8s.io/v1
     apiVersion: credentialprovider.kubelet.k8s.io/v1
     # Argumentos para passar ao comando quando for executá-lo.

>      # +opcional

Reviewer: `+optional` is a Go comment tag, so it should not be translated. See https://github.com/kubernetes/community/blob/8b47d07727de4004b78394d45afd84e03ac9d7b4/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required.

     args:
       - get-credentials

>      # Env define variáveis de ambiente adicionais para expor ao processo. Esses
>      # são unidos com o ambiente do host, bem como as variáveis que o client-go usa

Reviewer: ```suggestion
     # Env define variáveis de ambiente adicionais para expor ao processo. Esses valores
     # são combinados com o ambiente do host, bem como as variáveis que o client-go usa
```

Suggestion for readability.

     # para passar o argumento para o plugin.

>      # +opcional

Reviewer: Same consideration here about `+optional`.
